connectors/citrus-openapi/src/test/java/org/citrusframework/openapi/integration/OpenApiClientIT.java

```diff
@@ -17,83 +17,209 @@
+
+    @CitrusTest
+    @Test
+    public void shouldFailOnMissingNameInResponse() {
+
+        variable("petId", "1001");
+
+        when(openapi("petstore-v3")
+            .client(httpClient)
+            .send("getPetById")
+            .autoFill(AutoFillType.ALL)
+            .message()
+            .fork(true));
+
+        then(http().server(httpServer)
+            .receive()
+            .get("/pet/${petId}")
+            .message()
+            .accept("@contains('application/json')@"));
 
+        then(http().server(httpServer)
+            .send()
+            .response(OK)
+            .message()
+            .body(Resources.create(INVALID_PET_PATH))
+            .contentType(APPLICATION_JSON_VALUE));
+
+        OpenApiClientResponseActionBuilder clientResponseActionBuilder = openapi("petstore-v3")
+            .client(httpClient).receive("getPetById", OK)
+            .schemaValidation(true);
+        assertThrows(() -> then(clientResponseActionBuilder));
+    }
+
+    /**
+     * Tests validation of plain http client open api schema validation using
+     * {@link org.citrusframework.openapi.validation.OpenApiValidationContext}
+     */
     @CitrusTest
-    public void getPetById() {
+    @Test
+    public void shouldFailOnMissingNameInResponseWhenUsingExplicitSchemaValidation() {
         variable("petId", "1001");
 
-        when(openapi(petstoreSpec)
-                .client(httpClient)
-                .send("getPetById")
-                .fork(true));
+        when(http()
+            .client(httpClient)
+            .send()
+            .get("/pet/1001")
+            .message()
+            .fork(true));
 
         then(http().server(httpServer)
-                .receive()
-                .get("/pet/${petId}")
-                .message()
-                .accept("@contains('application/json')@"));
+            .receive()
+            .get("/pet/${petId}")
+            .message()
+            .accept("@contains('application/json')@"));
 
         then(http().server(httpServer)
-                .send()
-                .response(HttpStatus.OK)
-                .message()
-                .body(Resources.create("classpath:org/citrusframework/openapi/petstore/pet.json"))
-                .contentType("application/json"));
-
-        then(openapi(petstoreSpec)
-                .client(httpClient)
-                .receive("getPetById", HttpStatus.OK));
+            .send()
+            .response(OK)
+            .message()
+            .body(Resources.create(INVALID_PET_PATH))
+            .contentType(APPLICATION_JSON_VALUE));
+
+        // We can validate an OpenAPI response against a specific schema,
+        // even without using open api client.
+        HttpClientResponseActionBuilder.HttpMessageBuilderSupport receiveGetPetById = http().client(
+                httpClient)
+            .receive()
+            .response(OK)
+            .message()
+            .type(MessageType.JSON)
+            .validate(openApi(null)
```

Comment:
Having to pass `null` as an argument is not very comfortable. Please think of adding an empty argument method for this. Just to be sure that I get it right. Using the `null` argument here would perform a `normal` schema validation without the OpenAPI spec being included?

Reply:
Method was already there, wating for being used.

But no. There is no normal schema validation. For this, we would have to register the individual components of the OpenApi as json schemas - i guess. I don't know if there is a use case for this. The implementation of  OpenApi SchemaValidation follows the pattern of JsonSchemaValidation. Get repository by name and resolve schema from it. Therefore, it is not required to pass in an OpenApiSpecification. If you do so, the default enablement of the context would be derived from that spec. So for the validation to succeed, an OpenApiSpec needs to be registered.